### ISSUE TITLE
containerd: k8s-copy-file is flaky

### ISSUE THREAD

**Issue by fidencio** [kata-containers/tests#3061]
This bug reporthas been created based on: https://github.com/kata-containers/tests/pull/3060#issuecomment-726878427, where we could see a failure in the "Copy from pod to host" test.

As we solved a similar race condition on CRI-O recently, I'm opening this issue here as the containerd code may worth to be revisited.

/cc @liubin, @bergwolf

**Comment by wainersm** [comment]
I glanced at this test yesterday. It is really really simple, I am surprised there might be a race condition on it. :)

**Comment by liubin** [comment]
hi @fidencio @wainersm the original log  http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/186/console has gone, is there more details on this ?

**Comment by fidencio** [comment]
@liubin, yes, you can see the failure here, for instance: http://jenkins.katacontainers.io/job/jobs-2.0/job/kata-2-ubuntu-1804-PR-cloud-hypeprvisor-k8s-containerd/543/console

**Comment by fidencio** [comment]
```
02:01:19 ok 1 Copy file in a pod
02:01:21 not ok 2 Copy from pod to host
02:01:21 # (in test file k8s-copy-file.bats, line 50)
02:01:21 #   `cat "$file_name" | grep "$content"' failed
02:01:21 # INFO: k8s configured to use runtimeclass
02:01:21 # pod/test-env created
02:01:21 # pod/test-env condition met
02:01:21 # I1218 01:01:21.137180   15720 log.go:172] (0xc00056c160) (0xc0007fdcc0) Create stream
02:01:21 # I1218 01:01:21.137225   15720 log.go:172] (0xc00056c160) (0xc0007fdcc0) Stream added, broadcasting: 1
02:01:21 # I1218 01:01:21.138309   15720 log.go:172] (0xc00056c160) Reply frame received for 1
02:01:21 # I1218 01:01:21.138340   15720 log.go:172] (0xc00056c160) (0xc0007d4000) Create stream
02:01:21 # I1218 01:01:21.138348   15720 log.go:172] (0xc00056c160) (0xc0007d4000) Stream added, broadcasting: 3
02:01:21 # I1218 01:01:21.138911   15720 log.go:172] (0xc00056c160) Reply frame received for 3
02:01:21 # I1218 01:01:21.138940   15720 log.go:172] (0xc00056c160) (0xc0004b9400) Create stream
02:01:21 # I1218 01:01:21.138949   15720 log.go:172] (0xc00056c160) (0xc0004b9400) Stream added, broadcasting: 5
02:01:21 # I1218 01:01:21.139490   15720 log.go:172] (0xc00056c160) Reply frame received for 5
02:01:21 # I1218 01:01:21.155605   15720 log.go:172] (0xc00056c160) Data frame received for 5
02:01:21 # I1218 01:01:21.155628   15720 log.go:172] (0xc0004b9400) (5) Data frame handling
02:01:21 # I1218 01:01:21.159384   15720 log.go:172] (0xc00056c160) Data frame received for 1
02:01:21 # I1218 01:01:21.159405   15720 log.go:172] (0xc0007fdcc0) (1) Data frame handling
02:01:21 # I1218 01:01:21.159416   15720 log.go:172] (0xc00056c160) Data frame received for 3
02:01:21 # I1218 01:01:21.159432   15720 log.go:172] (0xc0007d4000) (3) Data frame handling
02:01:21 # I1218 01:01:21.159453   15720 log.go:172] (0xc0007fdcc0) (1) Data frame sent
02:01:21 # I1218 01:01:21.159475   15720 log.go:172] (0xc00056c160) (0xc0007fdcc0) Stream removed, broadcasting: 1
02:01:21 # I1218 01:01:21.159487   15720 log.go:172] (0xc00056c160) Go away received
02:01:21 # I1218 01:01:21.159695   15720 log.go:172] (0xc00056c160) (0xc0007fdcc0) Stream removed, broadcasting: 1
02:01:21 # I1218 01:01:21.159712   15720 log.go:172] (0xc00056c160) (0xc0007d4000) Stream removed, broadcasting: 3
02:01:21 # I1218 01:01:21.159719   15720 log.go:172] (0xc00056c160) (0xc0004b9400) Stream removed, broadcasting: 5
02:01:21 # I1218 01:01:21.261679   15735 log.go:172] (0xc0000ff080) (0xc000234780) Create stream
02:01:21 # I1218 01:01:21.261722   15735 log.go:172] (0xc0000ff080) (0xc000234780) Stream added, broadcasting: 1
02:01:21 # I1218 01:01:21.263041   15735 log.go:172] (0xc0000ff080) Reply frame received for 1
02:01:21 # I1218 01:01:21.263072   15735 log.go:172] (0xc0000ff080) (0xc0007d8aa0) Create stream
02:01:21 # I1218 01:01:21.263081   15735 log.go:172] (0xc0000ff080) (0xc0007d8aa0) Stream added, broadcasting: 3
02:01:21 # I1218 01:01:21.263662   15735 log.go:172] (0xc0000ff080) Reply frame received for 3
02:01:21 # I1218 01:01:21.263685   15735 log.go:172] (0xc0000ff080) (0xc000a3c000) Create stream
02:01:21 # I1218 01:01:21.263692   15735 log.go:172] (0xc0000ff080) (0xc000a3c000) Stream added, broadcasting: 5
02:01:21 # I1218 01:01:21.264316   15735 log.go:172] (0xc0000ff080) Reply frame received for 5
02:01:21 # I1218 01:01:21.279016   15735 log.go:172] (0xc0000ff080) Data frame received for 3
02:01:21 # I1218 01:01:21.279040   15735 log.go:172] (0xc0007d8aa0) (3) Data frame handling
02:01:21 # I1218 01:01:21.279051   15735 log.go:172] (0xc0007d8aa0) (3) Data frame sent
02:01:21 # cat: file.txt: No such file or directory
02:01:21 # pod "test-env" deleted
02:01:21 Failed at 86: bats "${K8S_TEST_ENTRY}"
```

**Comment by chavafg** [comment]
Tried to reproduce the error in an azure VM I spawned (same flavor as the once running in CI) and couldn't. I have changed the pod spec that it is being used for this test in #3131 and seems that with this change it is now working. Anyway it seems that there is still an issue when running the test with cloud-hypervisor, so it needs further debugging. In the meantime, we can 1) use #3131 or 2) skip the test for CLH. 

/cc @jcvenegas
